Flaky tests - Push notifications

cammellos opened the pull request:
Fixes some flaky push notification tests.
This was due to an actual bug in the implementation, a race condition that we thought we addressed but we were matching the wrong kind of error.
There still a know flaky test, that will be addressed separately.

cammellos committed:
bump version

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (3)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 825daff9 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2055/1/) | 2020-10-08 08:13:11 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-201008-081040-825daf.zip) |
| :heavy_check_mark: | 825daff9 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2055/1/) | 2020-10-08 08:14:10 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-201008-081037-825daf.zip) |
| :heavy_check_mark: | 825daff9 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2055/1/) | 2020-10-08 08:17:14 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-201008-081035-825daf.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | e6de33d0 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2055/2/) | 2020-10-08 10:49:49 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-201008-104749-e6de33.zip) |
| :heavy_check_mark: | e6de33d0 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2055/2/) | 2020-10-08 10:50:10 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-201008-104748-e6de33.zip) |
| :heavy_check_mark: | e6de33d0 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2055/2/) | 2020-10-08 10:53:03 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-201008-104745-e6de33.aar) |
| | | | | | | |
| :heavy_check_mark: | da6e9767 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2055/3/) | 2020-10-08 11:05:45 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-201008-110407-da6e97.zip) |
| :heavy_check_mark: | da6e9767 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2055/3/) | 2020-10-08 11:06:02 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-201008-110407-da6e97.zip) |
| :heavy_check_mark: | da6e9767 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2055/3/) | 2020-10-08 11:09:18 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-201008-110403-da6e97.aar) |

cammellos committed:
Use ErrRecordNotFound instead of sql.ErrNoRows

We were checking for the wrong error kind when pulling messages from the
database, which resulted in the code not retrying to pull the message,
giving flaky tests / race condition (that's present in production as
well)

cammellos committed:
Skip flaky test
